src/envelope/spacecharge/Elliptic_Integral.H

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2022-2023 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Chad Mitchell, Axel Huebl
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef ELLIPTIC_INTEGRAL_H
+#define ELLIPTIC_INTEGRAL_H
+
+#include <ablastr/warn_manager/WarnManager.H>
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Print.H>
+
+#include <cmath>
+
+namespace impactx::envelope::spacecharge
+{
+
+    /** Function to return the Carlson elliptic integral denoted
+     *  by RD(x,y,z).  This is a slight modification of
+     *  the C++ function written by John Burkardt (GNU LGPL license):
```

Comment:
Since we do not literally copy the source code, I would rephrase this as:
```suggestion
     *  by RD(x,y,z).  This is a reimplementation of
     *  the C++ function written by John Burkardt (GNU LGPL license):
```
and avoid license implications from copyleft.